File: bindings/lua/Polycode/SpriteState.lua
class "SpriteState"







function SpriteState:SpriteState(...)
	local arg = {...}
	for k,v in pairs(arg) do
		if type(v) == "table" then
			if v.__ptr ~= nil then
				arg[k] = v.__ptr
			end
		end
	end
	if self.__ptr == nil and arg[1] ~= "__skip_ptr__" then
		self.__ptr = Polycore.SpriteState(unpack(arg))
	end
end

function SpriteState:setName(name)
	local retVal = Polycore.SpriteState_setName(self.__ptr, name)
end

function SpriteState:getName()
	local retVal =  Polycore.SpriteState_getName(self.__ptr)
	return retVal
end

function SpriteState:getNumFrameIDs()
	local retVal =  Polycore.SpriteState_getNumFrameIDs(self.__ptr)
	return retVal
end

function SpriteState:getFrameIDAtIndex(index)
	local retVal = Polycore.SpriteState_getFrameIDAtIndex(self.__ptr, index)
	return retVal
end

function SpriteState:getMeshForFrameIndex(index)
	local retVal = Polycore.SpriteState_getMeshForFrameIndex(self.__ptr, index)
	if retVal == nil then return nil end
	local __c = _G["Mesh"]("__skip_ptr__")
	__c.__ptr = retVal
	return __c
end

function SpriteState:insertFrame(index, frameID)
	local retVal = Polycore.SpriteState_insertFrame(self.__ptr, index, frameID)
end

function SpriteState:removeFrameByIndex(frameIndex)
	local retVal = Polycore.SpriteState_removeFrameByIndex(self.__ptr, frameIndex)
end

function SpriteState:clearFrames()
	local retVal =  Polycore.SpriteState_clearFrames(self.__ptr)
end

function SpriteState:setPixelsPerUnit(ppu)
	local retVal = Polycore.SpriteState_setPixelsPerUnit(self.__ptr, ppu)
end

function SpriteState:getPixelsPerUnit()
	local retVal =  Polycore.SpriteState_getPixelsPerUnit(self.__ptr)
	return retVal
end

function SpriteState:rebuildStateMeshes()
	local retVal =  Polycore.SpriteState_rebuildStateMeshes(self.__ptr)
end

function SpriteState:setStateFPS(fps)
	local retVal = Polycore.SpriteState_setStateFPS(self.__ptr, fps)
end

function SpriteState:getStateFPS()
	local retVal =  Polycore.SpriteState_getStateFPS(self.__ptr)
	return retVal
end

function SpriteState:getLargestFrameBoundingBox()
	local retVal =  Polycore.SpriteState_getLargestFrameBoundingBox(self.__ptr)
	if retVal == nil then return nil end
	local __c = _G["Vector3"]("__skip_ptr__")
	__c.__ptr = retVal
	return __c
end

function SpriteState:setBoundingBox(boundingBox)
	local retVal = Polycore.SpriteState_setBoundingBox(self.__ptr, boundingBox.__ptr)
end

function SpriteState:getBoundingBox()
	local retVal =  Polycore.SpriteState_getBoundingBox(self.__ptr)
	if retVal == nil then return nil end
	local __c = _G["Vector2"]("__skip_ptr__")
	__c.__ptr = retVal
	return __c
end

function SpriteState:getSpriteOffset()
	local retVal =  Polycore.SpriteState_getSpriteOffset(self.__ptr)
	if retVal == nil then return nil end
	local __c = _G["Vector2"]("__skip_ptr__")
	__c.__ptr = retVal
	return __c
end

function SpriteState:setSpriteOffset(offset)
	local retVal = Polycore.SpriteState_setSpriteOffset(self.__ptr, offset.__ptr)
end

function SpriteState:__delete()
	if self then Polycore.delete_SpriteState(self.__ptr) end
end

File: bindings/javascript/Polycode/SpriteState.js
function SpriteState(spriteSet,name) {
	if(arguments[0] != "__skip_ptr__") {
		this.__ptr = Polycode.SpriteState(spriteSet,name)
	}
}


Duktape.fin(SpriteState.prototype, function (x) {
	if (x === SpriteState.prototype) {
		return;
	}
	Polycode.SpriteState__delete(x.__ptr)
})

SpriteState.prototype.setName = function(name) {
	Polycode.SpriteState_setName(this.__ptr, name)
}

SpriteState.prototype.getName = function() {
	return Polycode.SpriteState_getName(this.__ptr)
}

SpriteState.prototype.getNumFrameIDs = function() {
	return Polycode.SpriteState_getNumFrameIDs(this.__ptr)
}

SpriteState.prototype.getFrameIDAtIndex = function(index) {
	return Polycode.SpriteState_getFrameIDAtIndex(this.__ptr, index)
}

SpriteState.prototype.getMeshForFrameIndex = function(index) {
	var retVal = new Mesh()
	retVal.__ptr = Polycode.SpriteState_getMeshForFrameIndex(this.__ptr, index)
	return retVal
}

SpriteState.prototype.insertFrame = function(index,frameID) {
	Polycode.SpriteState_insertFrame(this.__ptr, index, frameID)
}

SpriteState.prototype.removeFrameByIndex = function(frameIndex) {
	Polycode.SpriteState_removeFrameByIndex(this.__ptr, frameIndex)
}

SpriteState.prototype.clearFrames = function() {
	Polycode.SpriteState_clearFrames(this.__ptr)
}

SpriteState.prototype.setPixelsPerUnit = function(ppu) {
	Polycode.SpriteState_setPixelsPerUnit(this.__ptr, ppu)
}

SpriteState.prototype.getPixelsPerUnit = function() {
	return Polycode.SpriteState_getPixelsPerUnit(this.__ptr)
}

SpriteState.prototype.rebuildStateMeshes = function() {
	Polycode.SpriteState_rebuildStateMeshes(this.__ptr)
}

SpriteState.prototype.setStateFPS = function(fps) {
	Polycode.SpriteState_setStateFPS(this.__ptr, fps)
}

SpriteState.prototype.getStateFPS = function() {
	return Polycode.SpriteState_getStateFPS(this.__ptr)
}

SpriteState.prototype.getLargestFrameBoundingBox = function() {
	var retVal = new Vector3()
	retVal.__ptr = Polycode.SpriteState_getLargestFrameBoundingBox(this.__ptr)
	return retVal
}

SpriteState.prototype.setBoundingBox = function(boundingBox) {
	Polycode.SpriteState_setBoundingBox(this.__ptr, boundingBox)
}

SpriteState.prototype.getBoundingBox = function() {
	var retVal = new Vector2()
	retVal.__ptr = Polycode.SpriteState_getBoundingBox(this.__ptr)
	return retVal
}

SpriteState.prototype.getSpriteOffset = function() {
	var retVal = new Vector2()
	retVal.__ptr = Polycode.SpriteState_getSpriteOffset(this.__ptr)
	return retVal
}

SpriteState.prototype.setSpriteOffset = function(offset) {
	Polycode.SpriteState_setSpriteOffset(this.__ptr, offset)
}
